api.bs

```diff
@@ -1701,27 +1701,39 @@ To perform <dfn>common matching logic</dfn>, given
 
 <div algorithm>
 To <dfn>fairly allocate credit</dfn>, given a [=list=] of doubles |credit| and an integer |value|:
+1.  [=Assert=] |credit| is not [=list/is empty|empty=].
 1.  Let |sumCredit| be the sum of |credit|'s [=list/items=].
-1.  Let |rawNormalizedCredit| be a new [=list=].
-1.  Let |normalizedCredit| be a new [=list=].
-1.  [=list/iterate|For each=] |item| of |credit|:
-      1. Let |rawNormalized| be |value| * |item| / |sumCredit|.
-      1. [=list/Append=] |rawNormalized| to |rawNormalizedCredit|.
-      1. Let |normalized| be |rawNormalized| rounded towards positive Infinity and converted to an integer.
-      1. [=list/Append=] |normalized| to |normalizedCredit|.
-1.  Let |shuffledFractionalIndices| be [=list/get the indices|the indices=] of |credit|, ordered randomly.
-1.  [=list/remove|Remove=] all indices |i| from |shuffledFractionalIndices| where |rawNormalizedCredit|[|i|] has no fractional part.
-1.  [=list/iterate|For each=] |index| of |shuffledFractionalIndices|:
-      1. If the sum of |normalizedCredit|'s [=list/items=] is equal to |value|, [=iteration/break=].
-      1. Decrement |normalizedCredit|[|index|] by 1.
-1.  Return |normalizedCredit|.
+1.  Let |unroundedCreditIndices| be a new [=list=].
+1.  Let |roundedCredit| be a new [=list=].
+1.  [=list/iterate|For each=] |idx| of [=list/get the indices|the indices=] of |credit|:
+    1.  Let |normalizedCredit| be |value| * |credit|[|idx|] / |sumCredit|.
+    1.  If |normalizedCredit| is not equal to |normalizedCredit| converted to an integer,
+        [=list/append=] |idx| to |unroundedCreditIndices|.
```

Comment:
With the resolution algorithm, you don't need to skip integer indices.  And it might be better not to have the additional timing side channel.

Reply:
Sorry I didn't quite follow this comment. Don't we need to skip integer indices to avoid divide-by-zero?